netci.groovy

```diff
@@ -1074,13 +1075,9 @@ def static getJobName(def configuration, def architecture, def os, def scenario,
 
 def static addNonPRTriggers(def job, def branch, def isPR, def architecture, def os, def configuration, def scenario, def isFlowJob, def isWindowsBuildOnlyJob, def bidailyCrossList) {
 
-    // Limited Windows ARM64 hardware is restricted for non-PR triggers to certain branches.
-    if (os == 'Windows_NT') {
-        if ((architecture == 'arm64') || (architecture == 'arm') || (architecture == 'armlb')) {
-            if (!(branch in Constants.WindowsArm64Branches)) {
-                return
-            }
-        }
+    // Limited hardware is restricted for non-PR triggers to certain branches.
+    if (jobRequiresLimitedHardware(architecture, os) && (!(branch in Constants.LimitedHardwareBranches))) {
+        return
```

Comment:
nit: shall we create `branchCanUseLimitedHardware(branch)` predicate and do something like?
```
if (jobRequiresLimitedHardware(architecture, os) && !branchCanUseLimitedHardware(branch))
    return
```

Reply:
It's a good idea; I'll try to remember that for a future change